File: data-jpa/src/main/java/top/bettercode/summer/data/jpa/query/SpecPredicate.java
package top.bettercode.summer.data.jpa.query;

import javax.persistence.criteria.CriteriaBuilder;
import javax.persistence.criteria.Predicate;
import javax.persistence.criteria.Root;

/**
 * @author Peter Wu
 */
public interface SpecPredicate<T, M extends SpecMatcher<T, M>> {

  Predicate toPredicate(Root<T> root, CriteriaBuilder criteriaBuilder);
}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/query/SpecPredicate.kt
package top.bettercode.summer.data.jpa.query

import jakarta.persistence.criteria.CriteriaBuilder
import jakarta.persistence.criteria.Predicate
import jakarta.persistence.criteria.Root

/**
 * @author Peter Wu
 */
interface SpecPredicate<T : Any?, M : SpecMatcher<T, M>> {
    fun toPredicate(root: Root<T>, criteriaBuilder: CriteriaBuilder): Predicate?
}
